### DISCUSSION TITLE
How to configure workers running at VMs behind PROXMOX

### DISCUSSION THREAD

**Discussion by eichher** [jhuckaby/Cronicle#827] [Q&A]
I want to run several workers at a PROXMOX virtualization host in the Cloud. Cronicle master is at the local network and I have only one IP address to access the PROXMOX host (but I can reach the VMs by configuring different ports via NAT). I installed a worker at one of the VMs and added it manually at the master server. I got connection, but the delievered ip address from the worker was the address of the PROXMOX internal network and server definition was greyed out. I changed the ip address manually in the json file of the master server, now the worker is active and can be used. But there is no way to add a second worker for the PROXMOX environment, because I have only a single ip address. I can redefine the worker port at the client VM in the config.json file, but I cannot add the worker to the master server, because the master server throws an error "Server is alreay in cluster". Is there a way to add a port definition for a worker at the master host?

**Comment by jhuckaby** [comment]
> Is there a way to add a port definition for a worker at the master host?

I'm so sorry, but there is currently no way to do that with Cronicle v1.

For Cronicle v2 (to be released in 2025), the workers don't have any port at all.  They don't run a web server, nor have incoming connections.  They themselves connect to the primary server, and never the other way around.

The idea is that this makes it much easier to accommodate complex network topologies, as the connections only happen in one direction.  The primary server is the only server that has to accept incoming connections.
